server/src/controllers/Events/resolver.ts

```diff
@@ -124,20 +139,43 @@ export class EventResolver {
           },
         },
       });
+      await prisma.user_event_roles.delete({
+        where: {
+          user_id_event_id_role_name: {
+            user_id: oldRsvp.user_id,
+            event_id: oldRsvp.event_id,
+            role_name: 'attendee',
+          },
+        },
+      });
 
       if (!event.invite_only && !oldRsvp.on_waitlist) {
         const waitingList = event.rsvps.filter((r) => r.on_waitlist);
 
-        if (waitingList.length > 0) {
+        if (waitingList.length) {
+          const acceptedRsvp = waitingList[0];
           await prisma.rsvps.update({
             where: {
               user_id_event_id: {
-                user_id: waitingList[0].user_id,
-                event_id: waitingList[0].event_id,
+                user_id: acceptedRsvp.user_id,
+                event_id: acceptedRsvp.event_id,
               },
             },
             data: { on_waitlist: false },
           });
+
+          const isSubscribed = acceptedRsvp.user.user_event_roles;
+
+          if (isSubscribed.length) {
```

Comment:
```suggestion
          const isSubscribed = acceptedRsvp.user.user_event_roles.length;

          if (isSubscribed) {
```
Nit: it's a bit surprising when an `isX` isn't just boolean.